dev/merge_arrow_pr.py

```diff
@@ -139,14 +164,24 @@ def __init__(self, jira_con, jira_id, project, cmd):
     def current_fix_versions(self):
         return self.issue.fields.fixVersions
 
+    @classmethod
+    def sort_versions(cls, versions):
+        # Use utility ComparableVersion class to sort JIRA text versions
+        comparable_versions = [
+            ComparableVersion(version) for version in versions
+        ]
+        comparable_versions = sorted(comparable_versions, reverse=True)
+        return [
+            comp_version.jira_version for comp_version in comparable_versions
+        ]
+
     def get_candidate_fix_versions(self, merge_branches=('master',)):
         # Only suggest versions starting with a number, like 0.x but not JS-0.x
         all_versions = self.jira_con.project_versions(self.project)
         unreleased_versions = [x for x in all_versions
                                if not x.raw['released']]
 
-        unreleased_versions = sorted(unreleased_versions,
-                                     key=lambda x: x.name, reverse=True)
+        unreleased_versions = self.sort_versions(unreleased_versions)
 
         mainline_versions = self._filter_mainline_versions(unreleased_versions)
```

Comment:
Let's move this before the sorting, so we don't have to deal with odd version numbers.